<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

## Footnotes

In [ ]:
while True:
    print('*', end='') #it was just the button on top lol i tried to ctrlc

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

In [4]:
print = 'PRINT'

In [1]:
print('Hello World')

Hello World


In [4]:
import math
import numpy as np

In [7]:
%%timeit

squares = [i**2 for i in range(10)]

660 ns ± 79 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [8]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

780 ns ± 52.7 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


The code was ran 1,000,000 times, 7 times.

780ns refers to the mean time taken within the 7 times to run the 1,000,000 loops.

$\pm$ 52.7 ns refers to the standard devitation within the 7 times to run the 1,000,000 loops.

where ns is nanoseconds

In [ ]:
print ([i**2 for i in range(10)])


squares = list()
for i in range(10):
    squares.append(i**2)
print (squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
